### DISCUSSION TITLE
Any solution to group a bulk jobs in one queue?

### DISCUSSION THREAD

**Discussion by orenKingK** [OptimalBits/bull#2582] [Q&A]
## Description
I have a pipeline and each pipeline contains several stages.
my problem is that I want to group bulk jobs added for stages.(connected to the same pipeline).
Instead of creating a Queue for each pipeline, I need a better solution.
In my opinion the solution would be Grouping somehow the stages to a pipeline or have parent-child relationship between jobs.

Please anyone?

**Comment by manast** [comment]
Not sure exactly what you are looking for but in BullMQ we have flows https://docs.bullmq.io/guide/flows
and in the Pro version, we have groups: https://docs.bullmq.io/bullmq-pro/groups

**Reply by orenKingK** [reply]
I'll try to give a better explanation. 

I add bulk jobs for the array of stages I need to execute one after another.

When I add another bulk jobs I want them to execute concurrtntly with the first bulk jobs added before. 

Right now the second bulk jobs wait for the first bulk jobs in order to start executing.

What should I do?

**Reply by manast** [reply]
How do you define bulk in this scenario?
